### DISCUSSION TITLE
Attach RunnableSequence to child run using RunTree

### DISCUSSION THREAD

**Discussion by LuaxY** [langchain-ai/langsmith-sdk#906] [Q&A] [answered]
Hi everyone,

I would like to know if it's possible to attach a `RunnableSequence` (and `RunnableParallel`) to a child created using RunTree API.

```py
def sub_taks(rt):
  child = rt.create_child(name="ChildName")
  
  llm = ChatOpenAI(...)
  prompt = ChatPromptTemplate.from_messages(...)
  chain = promtp | llm
  chain.invoke({...}, {'run_id': child.id}) # I would like to attach this RunnableSequence to my child run
  
  child.end(...)
  child.post()

# --------------------

rt = RunTree(
    name="CustomName",
    run_type="chain",
)

sub_task(rt)

rt.end(...)
rt.post()
```

I tried to set the `run_id` with `with_config({'run_id': child.id})` on the chain, and in the `invoke()` method, but both are not creating the RunnableSequence in my trace.

When using the top-level run ID (`rt.id` in my case), it's working, but it's adding them at the end, where I want them nested.

<img width="390" alt="image" src="https://github.com/user-attachments/assets/a949d582-5e2f-4db0-a03d-db92df9d880b">

Am I doing it wrong?

**Comment by hinthornw** [comment, marked as answer]
Could use

```python
with langsmith.tracing_context(parent=rt):
    chain.invoke(...)
```

**Reply by LuaxY** [reply]
Thank you! My RunTree is now working perfectly
